src/ngx-smart-modal/src/components/ngx-smart-modal.component.ts

```diff
@@ -118,9 +121,9 @@ export class NgxSmartModalComponent implements OnInit, OnDestroy {
       }
 
       this._changeDetectorRef.markForCheck();
-    });
+      this.onOpenFinished.emit(this);
+    }, this.hideDelay);
```

Comment:
That's not good at all. You're setting a delay for hiding to delay the opening of the modal. You shall find another way to do that.